src/theory/quantifiers/sygus/sygus_unif_strat.cpp

```diff
@@ -853,10 +884,80 @@ void SygusUnifStrategy::staticLearnRedundantOps(
         }
       }
     }
+    // do not use ITEs in occurrences of this type as a subtype of a condition
+    // enumerator type
+    if (etis->d_this == strat_ITE && restrictions.d_removeRetITEs)
+    {
+      const Datatype& dt =
+          static_cast<DatatypeType>(etn.toType()).getDatatype();
+      TermDbSygus* tds = d_qe->getTermDatabaseSygus();
+      for (std::pair<Node, NodeRole>& cec : etis->d_cenum)
+      {
+        // whether condition enumerator
+        if (cec.second != role_ite_condition)
+        {
+          continue;
+        }
+        // whether the condition has a subtype with an ITE with the same types
+        // as the one for the strategy. This would always be true for etn, if it
+        // is a subtype of the condition enumerator's type, but can also apply
+        // for other subtypes
+        std::vector<TypeNode> sf_types;
+        tds->getSubfieldTypes(cec.first.getType(), sf_types);
+        for (TypeNode& sf_tn : sf_types)
+        {
+          // check has ite
+          int ite_cons_index = tds->getKindConsNum(sf_tn, ITE);
+          if (ite_cons_index == -1)
+          {
+            continue;
+          }
+          const Datatype& dt_sf =
+              static_cast<DatatypeType>(sf_tn.toType()).getDatatype();
+          // check arg types match
+          unsigned i;
+          for (i = 0; i < 3; ++i)
+          {
+            if (tds->getArgType(dt[cindex], i)
```

Comment:
I dont think this check is correct. If i is 0, we should be checking that the arg type is equal to the overall type cec.first.getType(), for i=1,2, we are checking whether the arg type is equal to sf_tn.